xbmc/addons/gui/GUIDialogAddonInfo.cpp

```diff
@@ -158,7 +161,9 @@ void CGUIDialogAddonInfo::UpdateControls()
   if (!m_item)
     return;
 
-  bool isInstalled = NULL != m_localAddon.get();
+  bool isInstalled = CServiceBroker::GetAddonMgr().IsAddonInstalled(
+      m_item->GetAddonInfo()->ID(), m_item->GetAddonInfo()->Origin(),
+      m_item->GetAddonInfo()->Version());
```

Comment:
Not sure how expensive `m_item->GetAddonInfo()` is, but as you're calling it 3 times here I would introduce a variable to hold it's return value.